homework/schedule/schedule.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <chrono>
+#include <functional>
+#include <thread>
+#include <string>
+
+/* generic function */
+template<typename FUNCTION, typename DURATION, typename... ARGS>
+void schedule(FUNCTION function, DURATION time, ARGS... args) {
+    std::this_thread::sleep_for(time);
+    function(args...);
```

Comment:
use perfect forwarding here. std::forward<Args>(args)...

Reply:
Done. Indeed, looks better and works. Thanks MA!